### PR TITLE
SkeletonHorse Additions

### PR DESCRIPTION
**Pull request by BillyGalbreath**

### CODE REVIEW

File: Spigot-API-Patches/0127-SkeletonHorse-Additions.patch [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
++    }
++
++    @Override
++    public void setCancelled(boolean cancel) {
++        this.cancelled = cancel;
++    }
++
++    @Override
++    public HandlerList getHandlers() {
++        return null;
```
Comment (kashike) [inline Spigot-API-Patches/0127-SkeletonHorse-Additions.patch]: Oops!

File: Spigot-API-Patches/0127-SkeletonHorse-Additions.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
++
++import org.bukkit.entity.SkeletonHorse;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++
++/**
++ * Event called when a player gets close to a skeleton horse and triggers the lightning trap
++ */
++public class TrapHorseEvent extends EntityEvent implements Cancellable {
```
Comment (kashike) [inline Spigot-API-Patches/0127-SkeletonHorse-Additions.patch]: This needs a better name.

File: Spigot-API-Patches/0127-SkeletonHorse-Additions.patch [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+diff --git a/src/main/java/org/bukkit/entity/SkeletonHorse.java b/src/main/java/org/bukkit/entity/SkeletonHorse.java
+index b2c6b6a8..beb8d1ba 100644
+--- a/src/main/java/org/bukkit/entity/SkeletonHorse.java
++++ b/src/main/java/org/bukkit/entity/SkeletonHorse.java
+@@ -3,4 +3,12 @@ package org.bukkit.entity;
+ /**
+  * Represents a SkeletonHorse - variant of {@link AbstractHorse}.
+  */
+-public interface SkeletonHorse extends AbstractHorse { }
++// Paper start
```
Comment (kashike) [inline Spigot-API-Patches/0127-SkeletonHorse-Additions.patch]: Incorrect placement.

File: Spigot-API-Patches/0127-SkeletonHorse-Additions.patch [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
++public interface SkeletonHorse extends AbstractHorse {
++    int getTrapTime();
++
++    boolean isTrap();
++
++    void setTrap(boolean trap);
++}
++// Paper end
```
Comment (kashike) [inline Spigot-API-Patches/0127-SkeletonHorse-Additions.patch]: Incorrect placement.

**Review by kashike — CHANGES_REQUESTED** [review]

**Review by aikar — APPROVED** [review]

**Pull request merged**